ui/pages/self_serve_details.tsx

```diff
@@ -0,0 +1,75 @@
+import type { NextPage } from "next";
+import React, { useEffect, useState } from "react";
+import PageLayout from "../components/PageLayout";
+import { useRouter } from "next/router";
+import { Input } from "@canonical/react-components";
+import { kratos } from "../api/kratos";
+import { handleFlowError } from "../util/handleFlowError";
+import { SettingsFlow } from "@ory/client";
+import { AxiosError } from "axios";
+import { getFullName, IdentityTraits } from "../util/selfServeHelpers";
+
+const SelfServeDetails: NextPage = () => {
+  const [flow, setFlow] = useState<SettingsFlow>();
+
+  // Get ?flow=... from the URL
+  const router = useRouter();
+  const { return_to: returnTo, flow: flowId } = router.query;
+
+  useEffect(() => {
+    // If the router is not ready yet, or we already have a flow, do nothing.
+    if (!router.isReady || flow) {
+      return;
+    }
+
+    // If ?flow=... was in the URL, we fetch it
+    if (flowId) {
+      kratos
+        .getSettingsFlow({ id: String(flowId) })
+        .then((res) => {
+          setFlow(res.data);
+        })
+        .catch(handleFlowError("settings", setFlow));
+      return;
+    }
+
+    // Otherwise we initialize it
+    kratos
+      .createBrowserSettingsFlow({
+        returnTo: returnTo ? String(returnTo) : undefined,
+      })
+      .then(({ data }) => {
+        if (flowId !== data.id) {
+          window.history.replaceState(
+            null,
+            "",
+            `./self_serve_details?flow=${data.id}`,
+          );
+          router.query.flow = data.id;
+        }
+        setFlow(data);
+      })
+      .catch(handleFlowError("settings", setFlow))
+      .catch(async (err: AxiosError<string>) => {
+        if (err.response?.data.trim() === "Failed to create settings flow") {
```

Comment:
question: do we have to rely on this text? Shouldn't there be an error ID here? I may remember wrong :-/

Reply:
We can see in kratos/handlers.go that here is no other information available. Just an http status 500, but that would be too generic.